lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,31 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+  # Gives the existing lookup paths that are suffixes of the fully qualified path of `klass`.
+  # @param [Object] klass
+  # @return [Array<Symbol>]
+  def nested_lookup_paths(klass)
+    paths = klass.to_s.split('::').reverse.reduce([]) { |a, e| a + [[e] + Array(a.last)] }.reverse
+    paths.map! { |path| "#{path.join('::')}" }
+    paths.map! do |path|
```

Comment:
@beauby can you add comments after each line, explaining what it does -- so that if debugging of this needs to happen in the future, we don't need to go through the exercise you and @bf4 went through?